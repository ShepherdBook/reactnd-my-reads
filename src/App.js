import React, { Component } from "react";
import "./App.css";
import Bookshelf from "./Bookshelf";
import Search from "./Search";
import { Route } from "react-router-dom";

import * as BooksApi from "./BooksAPI";

class App extends Component {
  state = {
    booksOnShelf: []
  };

  componentDidMount() {
    BooksApi.getAll().then(books => {
      this.setState(() => ({
        booksOnShelf: books
      }));
    });
  }

  moveBook = (book, newShelfName) => {
    BooksApi.update(book, newShelfName).then();
  };

  render() {
    return (
      <div className="body">
        <Route
          exact
          path="/"
          render={() => (
            <div className="list-books-content">
              <Bookshelf
                className="bookshelf"
                books={this.state.booksOnShelf}
                handleChange={this.moveBook}
              />
            </div>
          )}
        />

        <Route
          exact
          path="/search"
          render={() => (
            <Search
              books={this.state.booksOnShelf}
              handleChange={this.moveBook}
            />
          )}
        />
      </div>
    );
  }
}

export default App;
