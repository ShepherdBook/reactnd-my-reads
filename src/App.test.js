import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("App renders as expected", () => {
  const app = renderer.create(<App />);

  let tree = app.toJSON();
  expect(tree).toMatchSnapshot();
});
