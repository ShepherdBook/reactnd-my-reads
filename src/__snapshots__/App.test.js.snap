// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<div
  className="body"
>
  <div
    className="list-books-content"
  >
    <div>
      <header
        className="list-books-title"
      >
        <h1>
          my-reads
        </h1>
      </header>
      <div>
        <header
          className="bookshelf-title"
        >
          <h2>
            Currently Reading
          </h2>
        </header>
        <ul
          className="books-grid"
        />
      </div>
      <div>
        <header
          className="bookshelf-title"
        >
          <h2>
            Want To Read
          </h2>
        </header>
        <ul
          className="books-grid"
        />
      </div>
      <div>
        <header
          className="bookshelf-title"
        >
          <h2>
            Read
          </h2>
        </header>
        <ul
          className="books-grid"
        />
      </div>
      <a
        className="open-search"
        href="/search"
        onClick={[Function]}
      >
        <button>
          Open Search
        </button>
      </a>
    </div>
  </div>
</div>
`;
